config.go

```diff
@@ -0,0 +1,290 @@
+		Usage:       "Add a new App configuration",
+		UsageText:   "stream-cli config new",
+		Description: "Add a new App configuration which can be used on further operations",
+
+		Action: func(ctx *cli.Context) error {
+			newConfig := newDefaultConfig()
+			err := survey.Ask(questions(), &newConfig)
+			if err != nil {
+				return err
+			}
+
+			return config.Add(newConfig)
+		},
+	}
+}
+
+func removeConfigCmd(config *Config) *cli.Command {
+	return &cli.Command{
+		Name:        "remove",
+		Usage:       "Remove an App configuration",
+		UsageText:   "stream-cli config remove <app to remove>",
+		Description: "Remove an App configuration. This operation is irrevocable",
+
+		Action: func(ctx *cli.Context) error {
+			if ctx.Args().Len() != 1 {
+				return errors.New("remove command requires 1 argument")
+			}
+			return config.Remove(ctx.Args().First())
+		},
+	}
+}
+
+func listConfigsCmd(config *Config) *cli.Command {
+	return &cli.Command{
+		Name:        "list",
+		Usage:       "List all configurations",
+		UsageText:   "stream-cli config list",
+		Description: "List all app configurations",
+
+		Action: func(_ *cli.Context) error {
+			t := tabby.New()
+			t.AddHeader("", "Name", "Access Key", "Secret Key", "Region")
+
+			for k, v := range config.appsConfig {
+				defaultApp := ""
+				if v.Default {
+					defaultApp = "(default)"
+				}
+
+				secret := fmt.Sprintf("**************%v", v.AccessSecretKey[len(v.AccessSecretKey)-4:])
+				t.AddLine(defaultApp, k, v.AccessKey, secret, v.URL)
+			}
+			t.Print()
+			return nil
+		},
+	}
+}
+
+func defaultConfigCmd(config *Config) *cli.Command {
+	cmd := &cli.Command{
+		Name:        "default",
+		Usage:       "Set a configuration as the default",
+		UsageText:   "stream-cli config default <name of the configuration>",
+		Description: "Set a configuration as the default, it will be used by default for all operations",
+
+		Action: func(ctx *cli.Context) error {
+			if ctx.Args().Len() != 1 {
+				return errors.New("default command requires 1 argument")
+			}
+			return config.SetDefault(ctx.Args().First())
+		},
+	}
+
+	return cmd
+}
+
+type appConfig struct {
+	Name            string `yaml:"-"`
+	AccessKey       string `yaml:"access-key"`
+	AccessSecretKey string `yaml:"access-secret-key"`
+	URL             string `yaml:"url"`
+	Default         bool   `yaml:"default,omitempty"`
+}
+
+func newDefaultConfig() appConfig {
+	return appConfig{
+		URL: defaultEdgeURL,
+	}
+}
+
+type Config struct {
+	appsConfig map[string]*appConfig
+	file       *os.File
+}
+
+func NewConfig() (*Config, error) {
+	d, err := os.UserHomeDir()
+	if err != nil {
+		return nil, fmt.Errorf("cannot get user's home directory: %v", err)
+	}
+
+	err = os.Mkdir(path.Join(d, configDir), 0755)
+	if err != nil && !os.IsExist(err) {
+		return nil, fmt.Errorf("cannot create config directory: %v", err)
+	}
+
+	filepath := path.Join(d, configDir, configFile)
+	file, err := os.OpenFile(filepath, os.O_RDWR|os.O_CREATE, 0644)
```

Comment:
don't forget closing

Reply:
it's closed in the caller, but this will change